.circleci/config.yml

```diff
@@ -0,0 +1,57 @@
+version: 2.1
+
+executors:
+  python38:
+    docker:
+      - image: circleci/python:3.8
+
+commands:
+  setup_tox:
+    description: "Install tox"
+    steps:
+      - run: pip install -U tox virtualenv tox-factor
+
+  restore_tox_cache:
+    description: "Restore .tox directory from previous runs for faster installs"
+    steps:
+      - restore_cache:
+          # In the cache key:
+          #   - .Environment.CIRCLE_JOB: We do separate tox environments by job name, so caching and restoring is
+          #                              much faster.
+          key: tox-cache-{{ .Environment.CIRCLE_JOB }}-{{ checksum "tox.ini" }}
```

Comment:
I think you will also need to checksum `dev-requirements.txt` and maybe other requirements files as well. I noticed there is an issue where tox won't detect updates to external requirements specified with `-r` or `-c`, so CI will be stuck on old versions or miss new deps. See https://github.com/tox-dev/tox/issues/149.

Reply:
Good point, i'll take a note to add this for #828 since this current PR is really to just configure *any* circle ci config

Reply:
Added dev-requirements.txt, we might end up wanting to rely on it a bit more heavily especially for dependencies that take a while to build like grpcio